app/src/main/java/com/azuresamples/msalandroidapp/MSGraphRequestWrapper.java

```diff
@@ -24,71 +24,76 @@
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.microsoft.graph.requests.GraphServiceClient;
 
 import org.json.JSONObject;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
 
 public class MSGraphRequestWrapper {
     private static final String TAG = MSGraphRequestWrapper.class.getSimpleName();
 
     // See: https://docs.microsoft.com/en-us/graph/deployments#microsoft-graph-and-graph-explorer-service-root-endpoints
     public static final String MS_GRAPH_ROOT_ENDPOINT = "https://graph.microsoft.com/";
+    public static final String MS_GRAPH_V1_ENDPOINT = MS_GRAPH_ROOT_ENDPOINT + "v1.0";
+    public static final String MS_GRAPH_BETA_ENDPOINT = MS_GRAPH_ROOT_ENDPOINT + "beta";
 
     /**
-     * Use Volley to make an HTTP request with
+     * Use Graph SDK to make an HTTP request with
      * 1) a given MSGraph resource URL
      * 2) an access token
      * to obtain MSGraph data.
      **/
-    public static void callGraphAPIUsingVolley(@NonNull final Context context,
-                                               @NonNull final String graphResourceUrl,
-                                               @NonNull final String accessToken,
-                                               @NonNull final Response.Listener<JSONObject> responseListener,
-                                               @NonNull final Response.ErrorListener errorListener) {
-        Log.d(TAG, "Starting volley request to graph");
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    @SuppressWarnings("unchecked")
+    public static CompletableFuture<String> callGraphAPI(@NonNull final String graphResourceUrl,
+                                                         @NonNull final String accessToken) {
+        Log.d(TAG, "Starting SDK request to graph");
 
         /* Make sure we have a token to send to graph */
         if (accessToken == null || accessToken.length() == 0) {
-            return;
+            return CompletableFuture.completedFuture(null);
         }
 
-        RequestQueue queue = Volley.newRequestQueue(context);
-        JSONObject parameters = new JSONObject();
+        StaticTokenAuthProvider authProvider =
+            new StaticTokenAuthProvider(accessToken);
 
-        try {
-            parameters.put("key", "value");
-        } catch (Exception e) {
-            Log.d(TAG, "Failed to put parameters: " + e.toString());
-        }
+        GraphServiceClient graphClient = GraphServiceClient.builder()
+            .authenticationProvider(authProvider)
+            .buildClient();
 
-        JsonObjectRequest request = new JsonObjectRequest(Request.Method.GET, graphResourceUrl,
-                parameters, responseListener, errorListener) {
-            @Override
-            public Map<String, String> getHeaders() {
-                Map<String, String> headers = new HashMap<>();
-                headers.put("Authorization", "Bearer " + accessToken);
```

Comment:
The purpose of doing this is to show how exactly the token can be acquired and use to access *any* resources 
(We picked Graph because Graph's User.Read is accessible to every accounts).

This change would encapsulate that.

I'd suggest we have an extra page for Graph SDK, and you're free to bring in all the best practices with the Graph SDK in that one.

Reply:
(If all they need to access MSGraph, and Graph SDK already wraps around the authentication part, I'd say we probably won't even need MSAL in that page - cc @hamiltonha)

Reply:
Graph SDK does not "do" authentication - it relies on an auth provider. It can use the provider from the Azure Identity SDK, but that SDK isn't supported in Android. So, I wrote a custom provider that wraps the token you guys already get from MSAL.

My goal here was not to undo any of the token acquisition work you've already done, just to remove the HTTP calls to Graph (which we do not want to encourage). cc @baywet

Reply:
@rpdome I think there are multiple conflicting goals here:
- Showcase how to call a custom API with native http client
- Showcase how to call Microsoft Graph with code that is closer to real life for customers

We, as Microsoft, don't want to encourage people to call Microsoft Graph with a native client. Of course they technically can do it, but they shouldn't as they as loosing a lot of value from the different handlers to the types and the fluent APIs. Not using the SDK to call Microsoft Graph makes their applications less resilient and is not what is recommended.

Now, I understand you also want to showcase calling an API using the native HTTP client, this is fine, just pick a different API that doesn't have an SDK, or send the request to localhost with a mock.

I hope that makes sense and we can merge this PR in? We're on the identity call tomorrow and it'd be great if things were aligned.

Reply:
In that case, what about having both options in the code, but comment out the manual part? 

We can explicitly call out something along the line of "If you're using MSGraph, Please use the SDK. The manual way is kept here - in case you're trying to call other APIs"

We don't want to mock the request - because we still want the customer to 'play with the real thing' - this sample app is supposed to be a playground after all, and MSGraph is the only SDK that are available to everyone by default.